### PR TITLE
Fix unquoted paths.

### PR DESCRIPTION
**Pull request by designatednerd**
Addresses Vokal fork issue #4.

### CODE REVIEW

**Comment by designatednerd** [comment]
@chillpop Let me know if this fixes the issues on your project.

**Comment by seanwolter** [comment]
LGTM

File: envcov.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ then
 fi
 
 LCOV_PATH="${XCODECOVERAGE_PATH}/lcov-1.11/bin"
-OBJ_DIR=${OBJECT_FILE_DIR_normal}/${CURRENT_ARCH}
+OBJ_DIR="${OBJECT_FILE_DIR_normal}/${CURRENT_ARCH}"
 
 # Fix for the new LLVM-COV that requires gcov to have a -v paramter
 LCOV() {
```
Comment (chillpop) [inline envcov.sh]: I think the line below this (40) also needs some quotes

**Comment by chillpop** [comment]
Trying to completely fix this may not be worth it.  Even with these changes (and the additional line I mentioned above) I'm still getting path errors from geninfo:
`
geninfo: Can't exec "/Users/chp/vokal/Thousand": No such file or directory at /Users/chp/vokal/Thousand Trails/ThousandTrails-iOS/ThousandTrails/Pods/XcodeCoverage/lcov-1.11/bin/geninfo line 1901.
`
Do we want to be debugging ~3700 line perl scripts? Or should we just include a disclaimer that you can't have spaces in your path if you want to run this?

**Comment by designatednerd** [comment]
I mean, we should at least try to have the bits we've worked on be properly quoted, but no, I do not want to be debugging 3700 line perl scripts. 

I'll add quotes to the bit you suggested and a note to the README.

**Comment by chillpop** [comment]
JK, it's a really easy fix:
change line 1901 in the `geninfo` script from:
`open(GCOV_PIPE, "-|", "$gcov_tool --version")`
to:
`open(GCOV_PIPE, "-|", "\"$gcov_tool\" --version")`
(add escaped quotes around `$gcov_tool`)

**Comment by chillpop** [comment]
oh, and while this PR is open, could you add these lines to `getcov`:
`    # remove mogenerator files
    LCOV --remove ${LCOV_INFO} "_*.m" -d "${OBJ_DIR}" -o ${LCOV_INFO}
`
in the `exclude_data` function to ignore mogenerated managed object classes?

**Comment by designatednerd** [comment]
@chillpop Oh nice, glad I got distracted for 10 minutes before addressing that. :smile: 

In terms of your `exclude_data` request, that's making an assumption that anything else anyone uses this for will only be using `_*.m` notation for MOGenerated files - unfortunately, I don't think that's a safe assumption.

File: envcov.sh (hunk 1 above)
Comment (designatednerd) [inline envcov.sh]: Actually, looking at line 40 - `XCODECOVERAGE_PATH` and `LCOV_PATH` are both quoted when generated, so we shouldn't need quotes around either of those for this to work.

Comment (chillpop) [inline envcov.sh]: You still need them.  The quotes when the variables are set don't carry over to when they are used.  (I'm trying all this locally with a path that has spaces, trust me.)  Bash variable quoting is weird.  I think the logic is that double quotes are basically used once and then disappear (more or less).

Comment (designatednerd) [inline envcov.sh]: Blargh. DAMMIT BASH

**Comment by chillpop** [comment]
LGTM

File: envcov.sh (hunk 1 above)
Comment (designatednerd) [inline envcov.sh]: (fix pushed)

**Comment by designatednerd** [comment]
@seanwolter @vokal-isaac @jacobvanorder Any additional thoughts or should I merge this down and tag it?

**Comment by seanwolter** [comment]
LGTM

**Comment by vokal-isaac** [comment]
I'd like to take a look, but I can't right now.

**Comment by designatednerd** [comment]
@vokal-isaac I'll hold off on merging down until you get a look.

File: envcov.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ ENV_DIR="${DIR}"
 if [[ $DIR == *Pods/XcodeCoverage* ]]
```
Comment (vokal-isaac) [inline envcov.sh]: I think the `[[ ... ]]` construct is bash-specific, so maybe add `#!/bin/bash` or some such at the top?

Comment (vokal-isaac) [inline envcov.sh]: Also, should `$DIR` here be `"`-quoted?

File: envcov.sh (hunk 1 above)
Comment (vokal-isaac) [inline envcov.sh]: Yeah, the quoting is during the act of setting, not on the set value.  When you then go to use the variable, its value is literally substituted, bringing back the whitespace issue.

File: envcov.sh [hunk 3]
```diff
@@ -33,9 +33,9 @@ then
 fi
```
Comment (vokal-isaac) [inline envcov.sh]: There's another unquoted `$DIR` in the `if` just above here, too.

**Comment by designatednerd** [comment]
@vokal-isaac Just pushed fixes for the stuff you pointed out 2 weeks ago. :stuck_out_tongue_closed_eyes:

**Comment by vokal-isaac** [comment]
LGTM.  What're you doing making changes while on PTO?

**Comment by designatednerd** [comment]
Waiting for people to finish packing so we could get to the airport on
time...for our flight that is delayed 1.5 hours.
-ES

Hand-delivered by the Mobile Internet Gnomes

---
Ellen Shapiro
c. 310-482-1286

On Mar 2, 2015, at 9:18 AM, Isaac <notifications@github.com> wrote:

LGTM. What're you doing making changes while on PTO?

—
Reply to this email directly or view it on GitHub
<https://github.com/vokal/XcodeCoverage/pull/5#issuecomment-76742176>.

**Pull request merged**